format.ts

```diff
@@ -1,36 +1,135 @@
-#!/usr/bin/env deno --allow-run
+#!/usr/bin/env deno --allow-run --allow-write
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+/**
+ * This script formats the source files in the repository.
+ *
+ * Usage: deno format.ts [--check]
+ *
+ * Options:
+ *   --check          Checks if the source files are formatted.
+ */
+import { args, platform, readAll, exit, run, readFile, writeFile } from "deno";
+import { parse } from "./flags/mod.ts";
+import { prettier, prettierPlugins } from "./prettier/prettier.ts";
 
-import { readAll, exit, run } from "deno";
+const encoder = new TextEncoder();
+const decoder = new TextDecoder();
 
-async function checkVersion() {
-  const prettierVersion = run({
-    args: ["bash", "-c", "prettier --version"],
-    stdout: "piped"
+// Runs commands in cross-platform way
+function xrun(opts) {
+  return run({
+    ...opts,
+    args: platform.os === "win" ? ["cmd.exe", "/c", ...opts.args] : opts.args
   });
-  const b = await readAll(prettierVersion.stdout);
-  const s = await prettierVersion.status();
-  if (s.code != 0) {
-    console.log("error calling prettier --version error");
-    exit(s.code);
+}
+
+// Gets the source files in the repository
+async function getSourceFiles() {
+  return decoder
+    .decode(
+      await readAll(
+        xrun({
+          args: ["git", "ls-files"],
```

Comment:
One thing to note: This includes deleted files. I don't think it's quite right, but I haven't been able to work out the correct modification...

Reply:
Thanks for pointing! I'll try fix this in another PR.

Reply:
Perhaps there ought to be recursive and name only options for readDir in deno!

Reply:
Other than this issue, I think this format file could be used as is by external projects.